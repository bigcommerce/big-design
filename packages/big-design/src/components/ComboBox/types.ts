import { Placement } from '@popperjs/core';
import React, { ComponentPropsWithoutRef, RefObject } from 'react';

import { InputProps } from '../Input';

interface BaseComboBox extends Omit<ComponentPropsWithoutRef<'input'>, 'children' | 'value'> {
  action?: ComboBoxAction;
  autoComplete?: string;
  autoWidth?: boolean;
  description?: React.ReactChild;
  disabled?: boolean;
  error?: InputProps['error'];
  filterable?: boolean;
  inputRef?: RefObject<HTMLInputElement> | React.Ref<HTMLInputElement>;
  label?: React.ReactChild;
  labelId?: string;
  maxHeight?: number;
  name?: string;
  optionsLoading?: boolean;
  placement?: Placement;
  positionFixed?: boolean;
  required?: boolean;
  onClose?(): void;
  onInputChange?(value: string): void;
  onOpen?(): void;
  onScrollToBottom?(): void;
}

export interface ComboBoxProps<T> extends BaseComboBox {
  options: Array<ComboBoxOption<T>> | Array<ComboBoxOptionGroup<T>>;
  value?: T;
  onOptionChange(value?: T, option?: ComboBoxOption<T>): void;
}

interface BaseItem extends Omit<ComponentPropsWithoutRef<'li'>, 'value'> {
  content: string;
  description?: string;
  disabled?: boolean;
  icon?: React.ReactElement;
  tooltip?: string;
}

export interface ComboBoxOption<T> extends BaseItem {
  value: T;
}

export interface ComboBoxAction extends BaseItem {
  actionType?: 'normal' | 'destructive';
  onActionClick(inputText: string | null): void;
}

export interface ComboBoxOptionGroup<T> {
  label: string;
  separated?: boolean;
  options: Array<ComboBoxOption<T>>;
}
