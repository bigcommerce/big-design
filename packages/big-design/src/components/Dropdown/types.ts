import { Placement } from '@popperjs/core';
import { ComponentPropsWithoutRef, ReactElement } from 'react';

export interface DropdownProps extends Omit<ComponentPropsWithoutRef<'ul'>, 'children'> {
  autoWidth?: boolean;
  disabled?: boolean;
  items: Array<DropdownItem | DropdownLinkItem> | DropdownItemGroup[];
  selectedItem?: DropdownItem;
  maxHeight?: number;
  placement?: Placement;
  positionFixed?: boolean;
  toggle: ReactElement<{ id?: string }>;
}

interface BaseItem extends Omit<ComponentPropsWithoutRef<'li'>, 'value'> {
  actionType?: 'normal' | 'destructive';
  content: string;
  description?: string;
  disabled?: boolean;
  icon?: ReactElement;
  tooltip?: string;
}

export interface DropdownItem extends BaseItem {
  hash?: string;
  type?: 'text';
  onItemClick(item: DropdownItem): void;
}

export interface DropdownLinkItem extends BaseItem {
  target?: HTMLAnchorElement['target'];
  type: 'link';
  url: string;
}

export interface DropdownItemGroup {
  label?: string;
  separated?: boolean;
  items: Array<DropdownItem | DropdownLinkItem>;
}
