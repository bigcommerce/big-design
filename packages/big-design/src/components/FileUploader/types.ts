import { MarginProps } from '../../helpers';
import { ButtonProps } from '../Button';

export interface Localization {
  upload: string;
  showLess: string;
  showMore: string;
  canNotUploadTheseFiles: string;
}

export interface ValidatorConfig {
  message?: string;
  type: string;
  validator: (file: File) => boolean;
}

export interface Action
  extends Omit<
    ButtonProps,
    'variant' | 'children' | 'iconOnly' | 'iconRight' | 'iconLeft' | 'isLoading' | keyof MarginProps
  > {
  label: string;
}
