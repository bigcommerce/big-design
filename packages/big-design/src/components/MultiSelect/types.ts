import { Placement } from '@popperjs/core';
import React, { ComponentPropsWithoutRef, RefObject } from 'react';

import { InputProps } from '../Input';
import { SelectAction, SelectOption } from '../Select';
import { SelectOptionGroup } from '../Select/types';

interface BaseSelect extends Omit<ComponentPropsWithoutRef<'input'>, 'children' | 'value'> {
  action?: SelectAction;
  autoComplete?: string;
  autoWidth?: boolean;
  description?: React.ReactNode;
  disabled?: boolean;
  error?: InputProps['error'];
  filterable?: boolean;
  inputRef?: RefObject<HTMLInputElement> | React.Ref<HTMLInputElement>;
  label?: React.ReactNode;
  labelId?: string;
  maxHeight?: number;
  name?: string;
  placement?: Placement;
  positionFixed?: boolean;
  required?: boolean;
  onClose?(): void;
  onOpen?(): void;
}

export interface MultiSelectLocalization {
  selectAll: string;
}

export interface MultiSelectProps<T> extends BaseSelect {
  options: Array<SelectOption<T>> | Array<SelectOptionGroup<T>>;
  value?: T[];
  selectAll?: boolean;
  localization?: MultiSelectLocalization;
  onOptionsChange(value: T[], option: Array<SelectOption<T>>): void;
}
