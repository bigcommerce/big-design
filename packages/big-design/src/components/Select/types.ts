import { Placement } from '@popperjs/core';
import React, { ComponentPropsWithoutRef, RefObject } from 'react';

import { InputProps } from '../Input';

interface BaseSelect extends Omit<ComponentPropsWithoutRef<'input'>, 'children' | 'value'> {
  action?: SelectAction;
  autoComplete?: string;
  autoWidth?: boolean;
  description?: React.ReactChild;
  disabled?: boolean;
  error?: InputProps['error'];
  filterable?: boolean;
  inputRef?: RefObject<HTMLInputElement> | React.Ref<HTMLInputElement>;
  label?: React.ReactChild;
  labelId?: string;
  maxHeight?: number;
  name?: string;
  placement?: Placement;
  positionFixed?: boolean;
  required?: boolean;
  onClose?(): void;
  onOpen?(): void;
}

export interface SelectProps<T> extends BaseSelect {
  options: Array<SelectOption<T>> | Array<SelectOptionGroup<T>>;
  value?: T;
  onOptionChange(value?: T, option?: SelectOption<T>): void;
}

interface BaseItem extends Omit<ComponentPropsWithoutRef<'li'>, 'value'> {
  content: string;
  description?: string;
  disabled?: boolean;
  icon?: React.ReactElement;
  tooltip?: string;
}

export interface SelectOption<T> extends BaseItem {
  value: T;
}

export interface SelectAction extends BaseItem {
  actionType?: 'normal' | 'destructive';
  onActionClick(inputText: string | null): void;
}

export interface SelectOptionGroup<T> {
  label: string;
  separated?: boolean;
  options: Array<SelectOption<T>>;
}
