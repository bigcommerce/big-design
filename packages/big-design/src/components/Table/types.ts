import { ComponentPropsWithoutRef, ReactNode } from 'react';

import { MarginProps } from '../../helpers';
import { OffsetPaginationProps } from '../OffsetPagination';
import { StatelessPaginationProps } from '../StatelessPagination';

import { TableColumnDisplayProps } from './helpers';

export interface TableSelectable<T> {
  selectedItems: T[];
  onSelectionChange(selectedItems: T[]): void;
}

export type TableSortDirection = 'ASC' | 'DESC';

export interface TableSortable<T> {
  direction: TableSortDirection;
  columnHash?: string;
  onSort(columnHash: string, direction: TableSortDirection, column: TableColumn<T>): void;
}

export interface TableItem {
  id?: string | number;
  [key: string]: any;
}

export interface TableColumn<T> extends TableColumnDisplayProps {
  align?: 'left' | 'center' | 'right';
  hash: string;
  header: string;
  tooltip?: string;
  hideHeader?: boolean;
  isSortable?: boolean;
  render:
    | React.ComponentType<T & { children?: ReactNode }>
    | ((props: T & { children?: ReactNode }, context?: any) => string | number);
  verticalAlign?: 'top' | 'middle';
  width?: number | string;
  withPadding?: boolean;
}

type WithoutMarginProps<T> = Omit<T, keyof MarginProps>;

type StatelessPaginationPropsWithItemTotal = StatelessPaginationProps & { totalItems?: number };

export type DiscriminatedTablePaginationProps =
  | (WithoutMarginProps<OffsetPaginationProps> & { type: 'offset' })
  | (WithoutMarginProps<StatelessPaginationPropsWithItemTotal> & { type: 'stateless' });

export type TablePaginationProps =
  | WithoutMarginProps<OffsetPaginationProps>
  | WithoutMarginProps<StatelessPaginationPropsWithItemTotal>;

export interface Localization {
  ascendingOrder?: string;
  descendingOrder?: string;
  controlsLabel?: string;
}

export interface TableProps<T> extends ComponentPropsWithoutRef<'table'> {
  actions?: React.ReactNode;
  columns: Array<TableColumn<T>>;
  emptyComponent?: React.ReactElement;
  headerless?: boolean;
  itemName?: string;
  items: T[];
  keyField?: string;
  localization?: Localization;
  onRowDrop?(from: number, to: number): void;
  pagination?: TablePaginationProps;
  selectable?: TableSelectable<T>;
  sortable?: TableSortable<T>;
  stickyHeader?: boolean;
}
