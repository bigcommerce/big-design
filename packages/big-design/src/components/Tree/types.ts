import { RefObject } from 'react';

export type TreeNodeId = string;

export type TreeSelectableType = 'radio' | 'multi' | undefined;

export type TreeOnKeyDown<T> = (
  e: React.KeyboardEvent<HTMLLIElement>,
  node: {
    id: TreeNodeId;
    isExpanded: boolean;
    isSelectable: boolean;
    hasChildren: boolean;
    value: T;
  },
) => void;

export type TreeOnNodeClick = (e: React.MouseEvent<HTMLLIElement>, nodeId: TreeNodeId) => void;

export interface TreeNodeProps<T> {
  children?: Array<TreeNodeProps<T>>;
  icon?: React.ReactNode | null;
  id: TreeNodeId;
  label: string;
  value?: T;
}

export interface TreeExpandable {
  expandedNodes: TreeNodeId[];
  onCollapse?(nodeId: TreeNodeId): void;
  onExpand?(nodeId: TreeNodeId): void;
  onToggle?(nodeId: TreeNodeId, isExpanded: boolean): void;
}

export interface TreeSelectable<T> {
  selectedNodes?: TreeNodeId[];
  type: TreeSelectableType;
  onSelect?: (nodeId: TreeNodeId, value: T) => void;
}

export interface TreeFocusable {
  focusedNode: TreeNodeId;
  onFocus(nodeId: TreeNodeId): void;
}

export interface TreeProps<T> {
  nodes: Array<TreeNodeProps<T>>;
  iconless?: boolean;
  id?: string;
  expandable: TreeExpandable;
  focusable: TreeFocusable;
  selectable?: TreeSelectable<T>;
  disabledNodes?: TreeNodeId[];
  onKeyDown: TreeOnKeyDown<T>;
  onNodeClick?: TreeOnNodeClick;
}

export interface TreeContextState<T> {
  disabledNodes?: TreeNodeId[];
  expandable: TreeExpandable;
  focusable: TreeFocusable;
  iconless?: boolean;
  selectable?: TreeSelectable<T>;
  onKeyDown: TreeOnKeyDown<T>;
  onNodeClick?: TreeOnNodeClick;
  treeRef: RefObject<HTMLUListElement>;
}

export interface MapValues {
  children: TreeNodeId[];
  id: TreeNodeId;
  parent?: TreeNodeId;
}

export type NodeMap = Map<TreeNodeId, MapValues>;
