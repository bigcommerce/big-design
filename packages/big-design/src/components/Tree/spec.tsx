import { theme } from '@bigcommerce/big-design-theme';
import { fireEvent, render, RenderResult, screen } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';
import React from 'react';

import { Tree } from './Tree';
import { TreeExpandable, TreeFocusable, TreeNodeProps, TreeOnKeyDown, TreeProps } from './types';

const nodes: Array<TreeNodeProps<number>> = [
  {
    id: '0',
    value: 0,
    label: 'Test Node 0',
    children: [
      {
        id: '5',
        value: 5,
        label: 'Test Node 5',
        children: [{ id: '9', value: 9, label: 'Test Node 9' }],
      },
    ],
  },
  {
    id: '1',
    value: 1,
    label: 'Test Node 1',
    children: [{ id: '6', value: 6, label: 'Test Node 6' }],
  },
  { id: '2', value: 2, label: 'Category' },
  {
    id: '3',
    value: 3,
    label: 'Test Node 3',
    children: [{ id: '7', value: 7, label: 'Test Node 7' }],
  },
  {
    id: '4',
    value: 4,
    label: 'Test Node 4',
    children: [{ id: '8', value: 8, label: 'Test Node 8' }],
  },
];

let expandable: TreeExpandable;
let focusable: TreeFocusable;
let onKeyDown: TreeOnKeyDown<unknown>;

beforeEach(() => {
  expandable = { expandedNodes: [], onToggle: jest.fn() };
  focusable = { focusedNode: '', onFocus: jest.fn() };
  onKeyDown = jest.fn();
});

const renderDefaultTree = (
  additionalProps?: Partial<TreeProps<unknown>>,
): RenderResult & {
  expandable: TreeExpandable;
  focusable: TreeFocusable;
  onKeyDown: TreeOnKeyDown<unknown>;
} => {
  const rendered = render(
    <Tree
      expandable={expandable}
      focusable={focusable}
      nodes={nodes}
      onKeyDown={onKeyDown}
      {...additionalProps}
    />,
  );

  return {
    ...rendered,
    expandable,
    focusable,
    onKeyDown,
  };
};

test('renders simple tree', async () => {
  renderDefaultTree();

  const tree = await screen.findByRole('tree');

  expect(tree).toMatchSnapshot();
});

test('does not forward styles', async () => {
  // eslint-disable-next-line
  // @ts-ignore
  renderDefaultTree({ className: 'test', style: { background: 'red' } });

  const tree = await screen.findByRole('tree');

  expect(tree).not.toHaveClass('test');
  expect(tree.getElementsByClassName('test')).toHaveLength(0);
  expect(tree).not.toHaveStyle('background: red');
});

test('expands node on click', async () => {
  const { expandable } = renderDefaultTree();
  const firstTreeNode = await screen.findByText('Test Node 0');

  await userEvent.click(firstTreeNode);

  expect(expandable.onToggle).toHaveBeenCalledWith('0', false);
});

test('calles onKeyDown event', async () => {
  const { onKeyDown } = renderDefaultTree();
  const firstTreeNode = await screen.findByRole('treeitem', { name: 'Test Node 0' });

  await userEvent.type(firstTreeNode, '{Enter}');

  expect(onKeyDown).toHaveBeenCalledWith(
    expect.anything(),
    expect.objectContaining({
      id: '0',
    }),
  );
});

test('renders with no icons', async () => {
  renderDefaultTree({
    nodes: [{ id: '0', label: 'Test Node 0' }],
    iconless: true,
  });

  const tree = await screen.findByRole('tree');

  expect(tree.querySelector('svg')).not.toBeInTheDocument();
});

test('renders node with custom icon', async () => {
  renderDefaultTree({
    nodes: [{ id: '0', icon: <span data-testid="test-id" />, label: 'Test Node 0' }],
  });

  const icon = await screen.findByTestId('test-id');

  expect(icon).toBeInTheDocument();
});

test('iconless renders without custom icons', () => {
  renderDefaultTree({
    nodes: [{ id: '0', icon: <span data-testid="test-id" />, label: 'Test Node 0' }],
    iconless: true,
  });

  const icon = screen.queryByTestId('test-id');

  expect(icon).not.toBeInTheDocument();
});

test('renders radio select', async () => {
  renderDefaultTree({
    nodes: [{ id: '0', value: 0, label: 'Test Node 0' }],
    selectable: { type: 'radio' },
  });

  const tree = await screen.findByRole('tree');
  const radio = tree.querySelector('label');

  expect(tree).toHaveAttribute('aria-multiselectable', 'false');
  expect(radio).toHaveStyle(`border-radius: ${theme.borderRadius.circle}`);
});

test('renders multiselect buttons', async () => {
  renderDefaultTree({
    nodes: [{ id: '0', value: 0, label: 'Test Node 0' }],
    selectable: { type: 'multi' },
  });

  const tree = await screen.findByRole('tree');
  const multi = tree.querySelector('label');

  expect(tree).toHaveAttribute('aria-multiselectable', 'true');
  expect(multi).toHaveStyle(`border-radius: ${theme.borderRadius.normal}`);
});

test('trigger onSelect', () => {
  const onSelect = jest.fn();
  const { container } = renderDefaultTree({
    nodes: [{ id: '0', value: 0, label: 'Test Node 0' }],
    selectable: { type: 'multi', onSelect },
  });

  const multi = container.querySelector('label');

  if (multi) {
    fireEvent.click(multi);

    expect(onSelect).toHaveBeenCalledWith('0', 0);
  }
});

test('renders expanded nodes', () => {
  renderDefaultTree({
    expandable: { expandedNodes: ['0', '5'], onExpand: jest.fn() },
  });

  const node5 = screen.getByText('Test Node 5');
  const node6 = screen.queryByText('Test Node 6');
  const node9 = screen.getByText('Test Node 9');

  expect(node5).toBeVisible();
  expect(node6).toBeNull();
  expect(node9).toBeVisible();
});

test("disabled nodes don't trigger onSelect", async () => {
  const onSelect = jest.fn();

  renderDefaultTree({
    nodes: [{ id: '0', value: 0, label: 'Test Node 0' }],
    selectable: { type: 'multi', onSelect },
    disabledNodes: ['0'],
  });

  const tree = await screen.findByRole('tree');
  const multi = tree.querySelector('label');

  if (multi) {
    fireEvent.click(multi);

    expect(onSelect).not.toHaveBeenCalled();
  }
});

test('triggers onNodeClick', async () => {
  const onNodeClick = jest.fn();

  renderDefaultTree({
    nodes: [{ id: '0', label: 'Test Node 0' }],
    onNodeClick,
  });

  const node = await screen.findByText('Test Node 0');

  await userEvent.click(node);

  expect(onNodeClick).toHaveBeenCalledTimes(1);
});

test('clicking node triggers onFocus', async () => {
  const onFocus = jest.fn();

  renderDefaultTree({
    nodes: [{ id: '0', label: 'Test Node 0' }],
    focusable: { focusedNode: '', onFocus },
  });

  const node = await screen.findByText('Test Node 0');

  await userEvent.click(node);

  expect(onFocus).toHaveBeenCalledTimes(1);
});

test('collapsing node triggers onCollapse', async () => {
  const onCollapse = jest.fn();

  renderDefaultTree({
    nodes: [{ id: '0', label: 'Test Node 0', children: [{ id: '1', label: 'Test Node 1' }] }],
    expandable: { expandedNodes: ['0'], onCollapse },
  });

  const node = await screen.findByText('Test Node 0');

  await userEvent.click(node);

  expect(onCollapse).toHaveBeenCalledWith('0');
});

test('expanding node triggers onExpand', async () => {
  const onExpand = jest.fn();

  renderDefaultTree({
    nodes: [{ id: '0', label: 'Test Node 0', children: [{ id: '1', label: 'Test Node 1' }] }],
    expandable: { expandedNodes: [], onExpand },
  });

  const node = await screen.findByText('Test Node 0');

  await userEvent.click(node);

  expect(onExpand).toHaveBeenCalledWith('0');
});

test('should focus when tabbed on', async () => {
  renderDefaultTree({
    nodes: [{ id: '0', label: 'Test Node 0' }],
    focusable: { focusedNode: '0', onFocus: jest.fn() },
  });

  const node = await screen.findByRole('treeitem');

  expect(node).not.toHaveFocus();

  await userEvent.tab();

  expect(node).toHaveFocus();
});
