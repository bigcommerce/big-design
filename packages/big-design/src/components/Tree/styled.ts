import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled from 'styled-components';

export const StyledUl = styled.ul<{ show?: boolean }>`
  list-style-type: none;
  margin: 0;
  padding: 0;
  user-select: none;

  > li > ul {
    padding-left: ${({ theme }) =>
      theme.helpers.addValues(theme.spacing.xLarge, theme.spacing.xxSmall)};
  }
`;

StyledUl.defaultProps = { theme: defaultTheme };
