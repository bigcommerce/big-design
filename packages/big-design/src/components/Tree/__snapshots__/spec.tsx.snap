// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders simple tree 1`] = `
<ul
  aria-multiselectable="false"
  class="styled__StyledUl-sc-3526f150-0 jXgIr"
  role="tree"
  style="overflow: hidden;"
>
  <li
    aria-expanded="false"
    class="styled__StyledLi-sc-912cb84-0 ddOgUf"
    role="treeitem"
    tabindex="-1"
  >
    <div
      class="styled__StyledBox-sc-12d4dbf-0 imudcb styled__StyledFlex-sc-3a1154b6-0 dNPsOo styled__StyledFlex-sc-912cb84-3 gknWxS"
    >
      <div
        class="styled__StyledBox-sc-12d4dbf-0 imudcb styled__StyledFlexItem-sc-2b88929e-0 jCGHbT styled__StyledArrowWrapper-sc-912cb84-1 lpeGLF"
      >
        <svg
          aria-hidden="true"
          class="base__StyledIcon-sc-a9u0e1-0 PHcub"
          color="secondary60"
          fill="currentColor"
          focusable="false"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M9.29 6.71a.996.996 0 0 0 0 1.41L13.17 12l-3.88 3.88a.996.996 0 1 0 1.41 1.41l4.59-4.59a.996.996 0 0 0 0-1.41L10.7 6.7c-.38-.38-1.02-.38-1.41.01"
          />
        </svg>
      </div>
      <div
        class="styled__StyledBox-sc-12d4dbf-0 jlecRx styled__StyledFlexItem-sc-2b88929e-0 jCGHbT styled__StyledFlexItem-sc-912cb84-4 eopNjs"
      >
        <svg
          aria-hidden="true"
          class="base__StyledIcon-sc-a9u0e1-0 cjbLvM"
          color="primary30"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M10.59 4.59C10.21 4.21 9.7 4 9.17 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8z"
          />
        </svg>
      </div>
      <span
        class="styled__StyledText-sc-218b28a9-5 styled__StyledText-sc-912cb84-6 ffENPD ebpimB"
        color="secondary70"
      >
        Test Node 0
      </span>
    </div>
  </li>
  <li
    aria-expanded="false"
    class="styled__StyledLi-sc-912cb84-0 ddOgUf"
    role="treeitem"
    tabindex="-1"
  >
    <div
      class="styled__StyledBox-sc-12d4dbf-0 imudcb styled__StyledFlex-sc-3a1154b6-0 dNPsOo styled__StyledFlex-sc-912cb84-3 gknWxS"
    >
      <div
        class="styled__StyledBox-sc-12d4dbf-0 imudcb styled__StyledFlexItem-sc-2b88929e-0 jCGHbT styled__StyledArrowWrapper-sc-912cb84-1 lpeGLF"
      >
        <svg
          aria-hidden="true"
          class="base__StyledIcon-sc-a9u0e1-0 PHcub"
          color="secondary60"
          fill="currentColor"
          focusable="false"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M9.29 6.71a.996.996 0 0 0 0 1.41L13.17 12l-3.88 3.88a.996.996 0 1 0 1.41 1.41l4.59-4.59a.996.996 0 0 0 0-1.41L10.7 6.7c-.38-.38-1.02-.38-1.41.01"
          />
        </svg>
      </div>
      <div
        class="styled__StyledBox-sc-12d4dbf-0 jlecRx styled__StyledFlexItem-sc-2b88929e-0 jCGHbT styled__StyledFlexItem-sc-912cb84-4 eopNjs"
      >
        <svg
          aria-hidden="true"
          class="base__StyledIcon-sc-a9u0e1-0 cjbLvM"
          color="primary30"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M10.59 4.59C10.21 4.21 9.7 4 9.17 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8z"
          />
        </svg>
      </div>
      <span
        class="styled__StyledText-sc-218b28a9-5 styled__StyledText-sc-912cb84-6 ffENPD ebpimB"
        color="secondary70"
      >
        Test Node 1
      </span>
    </div>
  </li>
  <li
    aria-expanded="false"
    class="styled__StyledLi-sc-912cb84-0 ddOgUf"
    role="treeitem"
    tabindex="-1"
  >
    <div
      class="styled__StyledBox-sc-12d4dbf-0 imudcb styled__StyledFlex-sc-3a1154b6-0 dNPsOo styled__StyledFlex-sc-912cb84-3 gknWxS"
    >
      <div
        class="styled__StyledGap-sc-912cb84-5 jrVihD"
      />
      <div
        class="styled__StyledBox-sc-12d4dbf-0 jlecRx styled__StyledFlexItem-sc-2b88929e-0 jCGHbT styled__StyledFlexItem-sc-912cb84-4 eopNjs"
      >
        <svg
          aria-hidden="true"
          class="base__StyledIcon-sc-a9u0e1-0 cjbLvM"
          color="primary30"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M10.59 4.59C10.21 4.21 9.7 4 9.17 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8z"
          />
        </svg>
      </div>
      <span
        class="styled__StyledText-sc-218b28a9-5 styled__StyledText-sc-912cb84-6 ffENPD ebpimB"
        color="secondary70"
      >
        Category
      </span>
    </div>
  </li>
  <li
    aria-expanded="false"
    class="styled__StyledLi-sc-912cb84-0 ddOgUf"
    role="treeitem"
    tabindex="-1"
  >
    <div
      class="styled__StyledBox-sc-12d4dbf-0 imudcb styled__StyledFlex-sc-3a1154b6-0 dNPsOo styled__StyledFlex-sc-912cb84-3 gknWxS"
    >
      <div
        class="styled__StyledBox-sc-12d4dbf-0 imudcb styled__StyledFlexItem-sc-2b88929e-0 jCGHbT styled__StyledArrowWrapper-sc-912cb84-1 lpeGLF"
      >
        <svg
          aria-hidden="true"
          class="base__StyledIcon-sc-a9u0e1-0 PHcub"
          color="secondary60"
          fill="currentColor"
          focusable="false"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M9.29 6.71a.996.996 0 0 0 0 1.41L13.17 12l-3.88 3.88a.996.996 0 1 0 1.41 1.41l4.59-4.59a.996.996 0 0 0 0-1.41L10.7 6.7c-.38-.38-1.02-.38-1.41.01"
          />
        </svg>
      </div>
      <div
        class="styled__StyledBox-sc-12d4dbf-0 jlecRx styled__StyledFlexItem-sc-2b88929e-0 jCGHbT styled__StyledFlexItem-sc-912cb84-4 eopNjs"
      >
        <svg
          aria-hidden="true"
          class="base__StyledIcon-sc-a9u0e1-0 cjbLvM"
          color="primary30"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M10.59 4.59C10.21 4.21 9.7 4 9.17 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8z"
          />
        </svg>
      </div>
      <span
        class="styled__StyledText-sc-218b28a9-5 styled__StyledText-sc-912cb84-6 ffENPD ebpimB"
        color="secondary70"
      >
        Test Node 3
      </span>
    </div>
  </li>
  <li
    aria-expanded="false"
    class="styled__StyledLi-sc-912cb84-0 ddOgUf"
    role="treeitem"
    tabindex="-1"
  >
    <div
      class="styled__StyledBox-sc-12d4dbf-0 imudcb styled__StyledFlex-sc-3a1154b6-0 dNPsOo styled__StyledFlex-sc-912cb84-3 gknWxS"
    >
      <div
        class="styled__StyledBox-sc-12d4dbf-0 imudcb styled__StyledFlexItem-sc-2b88929e-0 jCGHbT styled__StyledArrowWrapper-sc-912cb84-1 lpeGLF"
      >
        <svg
          aria-hidden="true"
          class="base__StyledIcon-sc-a9u0e1-0 PHcub"
          color="secondary60"
          fill="currentColor"
          focusable="false"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M9.29 6.71a.996.996 0 0 0 0 1.41L13.17 12l-3.88 3.88a.996.996 0 1 0 1.41 1.41l4.59-4.59a.996.996 0 0 0 0-1.41L10.7 6.7c-.38-.38-1.02-.38-1.41.01"
          />
        </svg>
      </div>
      <div
        class="styled__StyledBox-sc-12d4dbf-0 jlecRx styled__StyledFlexItem-sc-2b88929e-0 jCGHbT styled__StyledFlexItem-sc-912cb84-4 eopNjs"
      >
        <svg
          aria-hidden="true"
          class="base__StyledIcon-sc-a9u0e1-0 cjbLvM"
          color="primary30"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M10.59 4.59C10.21 4.21 9.7 4 9.17 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8z"
          />
        </svg>
      </div>
      <span
        class="styled__StyledText-sc-218b28a9-5 styled__StyledText-sc-912cb84-6 ffENPD ebpimB"
        color="secondary70"
      >
        Test Node 4
      </span>
    </div>
  </li>
</ul>
`;
